mitosheet/mitosheet/public/v3/sheet_functions/utils.py

```diff
@@ -0,0 +1,142 @@
+
+
+
+from typing import Callable, List, Optional, Tuple, TypeVar, Union
+import numpy as np
+from datetime import datetime, timedelta
+
+import pandas as pd
+
+from mitosheet.public.v3.rolling_range import RollingRange
+
+
+PrimitiveType = TypeVar('PrimitiveType', bound=Union[str, float, int, bool, datetime, timedelta])
+
+ResultType = Union[pd.Series, PrimitiveType]
+
+
+def __get_default_value_if_value_is_none_or_nan(value: PrimitiveType, default_value: PrimitiveType) -> PrimitiveType:
+    if value is None or (isinstance(value, float) and np.isnan(value)):
```

Comment:
Because all NaN values are float values. You can't just put any old object into np.isnan or it will error. Something can only be an NaN if it's a float (or... if it's a date time - but we'll address that when we get there!).